Source/missiles.cpp

```diff
@@ -83,22 +83,18 @@ bool CheckBlock(Point from, Point to)
 
 Monster *FindClosest(Point source, int rad)
 {
-	if (rad > 19)
-		rad = 19;
-
-	for (int i = 1; i < rad; i++) {
-		int k = CrawlNum[i];
-		int ck = k + 2;
-		for (auto j = static_cast<uint8_t>(CrawlTable[k]); j > 0; j--, ck += 2) {
-			Point target = source + Displacement { CrawlTable[ck - 1], CrawlTable[ck] };
-			if (!InDungeonBounds(target))
-				continue;
+	std::optional<Point> monsterPosition = FindClosestValidPosition(
+	    [&source](Point target) {
+		    // search for a monster with clear line of sight
+		    return InDungeonBounds(target) && dMonster[target.x][target.y] > 0 && !CheckBlock(source, target);
+	    },
+	    source, 1, rad);

```

Comment:
The check here looked to be only for ensuring the read into CrawlNum and CrawlTable didn't go out of bounds, given the new function handles larger radiuses I let it pass through (it would've been more correct to use 18 to match the old behaviour exactly).

It did seem useful to have the search radius as a parameter for this FindClosestMonster function and it didn't look critical that we match the behaviour exactly here, but I'm happy to update the call sites to use 18 instead.